idl/truncation_record.idl.hh

```diff
@@ -1,19 +0,0 @@
-/*
- * Copyright 2016-present ScyllaDB
- */
-
-/*
- * SPDX-License-Identifier: AGPL-3.0-or-later
- */
-
-#include "idl/replay_position.idl.hh"
-
-namespace db {
-
-struct truncation_record {
-    uint32_t magic;
-    std::vector<db::replay_position> positions;
-    db_clock::time_point time_stamp;
-};
-
-}
```

Comment:
This is fishy, how can we remove something from idl without impacting compatibility?

Unless it was there by mistake.

Reply:
Before the patch `truncation_record` struct was used as a result type for `get_truncation_record`, which was inlined in this PR. It wasn't used for serialization.

`truncation_record.idl.hh` was added a while back. Maybe we just forgot to delete it after some refactoring in the past. @elcallio